saml/config.go

```diff
@@ -1,105 +1,73 @@
 package saml
 
 import (
-	"crypto/rsa"
-	"crypto/tls"
 	"errors"
 	"fmt"
 	"net/url"
 	"time"
 
 	"github.com/hashicorp/cap/oidc"
-	"github.com/hashicorp/cap/saml/models/core"
 )
 
 var ErrInvalidTLSCert = errors.New("invalid tls certificate")
 
 type ValidUntilFunc func() time.Time
 
-type IDPConfig struct {
-	// SSOServiceURL is the SSO service endpoint where the authentication request
-	// should be sent to. The endpoint must match the location for the given
-	// service binding.
-	SSOServiceURL string
-
-	// TODO: Add more fields for the cert, crypto algorithm, etc.
-	// SLOServiceURL string
-}
-
 type Config struct {
-	// ServiceBinding defines the mechanism that should be used
-	// when exchanging messages between the requester (SP) and responder (IDP).
-	ServiceBinding core.ServiceBinding
-
-	// NameIDFormat controls how the users at the IDP are mapped to
-	// users at the SP during SSO.
-	NameIDFormat core.NameIDFormat
-
 	// AssertionConsumerServiceURL defines the endpoint at the SP where the IDP
-	// will redirect to with its authentication response.
+	// will redirect to with its authentication response. (required)
 	AssertionConsumerServiceURL *url.URL
 
-	// EntityID is a globaly unique identifier of the service provider.
+	// EntityID is a globaly unique identifier of the service provider. (required)
 	EntityID *url.URL
 
-	// Issuer is a globaly unique identifier of the identity provider.
+	// Issuer is a globaly unique identifier of the identity provider. (required)
 	Issuer *url.URL
 
-	// MetadataURL is the endpoint an IDP serves its metadata XML document.
+	// MetadataURL is the endpoint an IDP serves its metadata XML document. (required)
 	MetadataURL *url.URL
 
-	// ValidUntil is a function that defines until the generate service provider metadata
-	// document is valid.
+	// ValidUntil is a function that defines until the generated service provider metadata
+	// document is valid. (optional)
 	ValidUntil ValidUntilFunc
-
-	// Certificate is used to sign SAML Authentication Requests.
-	Certificate *tls.Certificate
```

Comment:
As we don't plan to sign Auth request in first place I removed it for now. We can add it back when implementing that feature.

Reply:
Sounds good to me 👍